boardswarm/src/utils.rs

```diff
@@ -1,5 +1,18 @@
 use tracing::warn;
 
+#[macro_export]
+/// Get the value of a bit at a specific index
+///
+/// ```
+/// assert_eq!(get_bit!(0x1, 0), 1);
+/// assert_eq!(get_bit!(0x1, 1), 0);
```

Comment:
boardswarm isn't a library crate (at least not atm)  so doctests won't get run; It would be good to split that into a seperate test